### PR TITLE
fix(Deck) - show conversation name and highlight link in deck integration

### LINKED ISSUE #10331: Deck card sharing - success message - include more obvious link, conversation name
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Summary
Context: When sharing a card within Deck to a Talk conversation via the "Post to a conversation" feature

Upon sharing a Deck card to a Talk conversation via ^, the success message contains a link to the conversation it was posted to. However, the link is rather hidden, i.e. the word `conversation` is the link, but with no text colour change, underline, etc to display to the user it's clickable. 

The only way I discovered it was a link was through looking at the [code](https://github.com/nextcloud/spreed/blob/251587f5261f7ed277c68fdb0735969b5bfbc4bc/src/deck.js#L48) for a [related issue](#10330).

**Describe the solution you'd like**
Would be lovely to make the link more visible/discoverable, and stretch goal to make the conversation room name appear in the success message, instead of the generic "conversation":

current: "Deck card has been posted to the selected `[conversation](link)`"
could be "Deck card has been posted to `[name-of-conversation](link)`

image
(ignore typo `selectedconversation` as described in #10330 )

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Fix #10331

### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
image | Screenshot from 2023-08-27 16-03-04

### 🚧 Tasks

- [ ] Visual check
- [ ] Code review

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not required
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

File: src/deck.js [hunk 1]
```diff
@@ -38,16 +40,22 @@ import { postRichObjectToConversation } from './services/messagesService.js'
...
+
+			const messageId = responsePostCard.value.data.ocs.data.id
+			const conversation = responseGetConversation.value.data.ocs.data.displayName
 			const targetUrl = generateUrl('/call/{token}#message_{messageId}', { token, messageId })
-			showSuccess(t('spreed', 'Deck card has been posted to the selected <a href="{link}">conversation</a>', {
-				link: targetUrl,
-			}), {
+
+			showSuccess(escapeHtml(t('spreed', 'Deck card has been posted to {conversation}'))
+				.replace(/\{conversation}/g, `<a target="_blank" class="external" href="${targetUrl}">${conversation} ↗</a>`),
```
Comment (nickvergessen) [inline src/deck.js:57]: The translation should not need `escapeHtml(`. However the conversation name should have it as that is user input.

**Review by nickvergessen — CHANGES_REQUESTED** [review]
See comment

File: src/deck.js (hunk 1 above)
Comment (Antreesy) [reply src/deck.js]: Fixed

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**